File: src/main/java/es/joaquin/ecommerce/domain/ProductRepository.java
package es.joaquin.ecommerce.domain;

import java.util.List;
import java.util.Optional;

import es.joaquin.ecommerce.domain.dto.ProductDto;

public interface ProductRepository {
	
	ProductDto createProduct(Product product);

	List<ProductDto> getProducts();

	Optional<ProductDto> getProduct(Long id);

	Boolean deleteProduct(Long id);

}

File: src/infraestructure/productRepository.js
import {Product} from "./models/Product.js";

const productRepository = ({}) => ({
  
  async getProducts() {
    return Product.getProducts();
  },

  async createProduct(product) {
    const productEntityDto = new ProductEntity(product);
    return productEntityDto.createProduct();
  },

  async getProduct(id) {
    const productEntityDto = await Product.getProduct(id);
    if (productEntityDto) {
      return productEntityDto;
    }
    return false;
  },

  async deleteProduct(product) {
    const productEntityDto = await ProductEntity.findById(product.id);
    await productEntityDto.deleteProduct();
  },

});

module.exports = productRepository;